### ISSUE TITLE
Proper 'move' between columns

### ISSUE THREAD

**Issue by CloudGlueIO** [sdorra/angular-dashboard-framework#16]
GREAT framework, excellent work, much appreciated.

Currently a move between sortable lists/columns is actually a delete from one list and an add to another.  
This results in the widget element/directive being re-generated as angular watch detects a change in the array.

This is is not ideal because:
a) Generally Inefficient
b) Widget visual state is lost

Being an Angular newbie, I'm not sure what the best way to handle this is.

I see a $$hashKey on the widget object, and that the object is spliced from one column to another, and I guess angular is watching and re-rendering the directive and the spliced widget's $$hashKey is no longer valid.

What can be done to support a pure move?

I inserted code to add an 'id' property to a widget instance, and 'track by definition.id' to the ng-repeat in dashboard.tpl.html.  Unfortunately it didn't make a difference.

```
<div class="widgets" ng-repeat="definition in col.widgets track by definition.id">
```

**Comment by sdorra** [comment]
I've create a new feature branch which uses [Sortable](https://github.com/RubaXa/Sortable) instead of jquery-ui. It seems to work very well, expect of one problem with nested columns. But the issue is already addressed [here](https://github.com/RubaXa/Sortable/pull/214).

- https://github.com/sdorra/angular-dashboard-framework/tree/feature/sortable

**Comment by xemware** [comment]
Awesome, look forward to the update.

**Comment by sdorra** [comment]
I was finally able to get dnd work without jquery-ui. I've used [sortable](https://github.com/RubaXa/Sortable) and implemented the dnd without the use of ng-sortable. This implementation should work even with [disabled debug mode](https://code.angularjs.org/1.3.15/docs/guide/production). The changes are merged into the master branch and will be available with the release of 0.8.0 of the adf.

* [c70b8b11](https://github.com/sdorra/angular-dashboard-framework/commit/c70b8b118c5d9db7e0bed257e455fce0be7b5fbf)

**Comment by TimNZ** [comment]
Great stuff, thanks for ongoing dev.
